### PR TITLE
Fix win installer

### PR DESCRIPTION
**Pull request by cbielow**
## Description

Fixes the Windows installer by:
1) ship VS runtime dll's in the ./bin directory (thus not requiring to install a vs_runtime.exe during install time -- which was disabled in the old installer anyway for some reason)
2) pwiz: 
    2.1) avoiding unneccessary .NET installs (which are not required for pwiz since any Win10 should ship them by default)
    2.2) avoiding unncessary .vs runtime installs since pwiz ships them in its own ./bin directory for a while now
3) force use of Qt5.15 when creating an installer, since Qt5.15 can find its qwindows.dll in ./bin/plugins/platforms on its own (using any prior version would crash any GUI tool with "The application failed to start ...  qt platform plugin "windows" "
image

upon start on the target machine)

Plus some minor doc fixes (FlashDeconvWizard was not mentioned in the docs).

these commits:
f4823ac7a20c694617e44724cdf25ac3ff228385
541a759ffa81f725ab141f263beafeeb16c85cbf
302b69fc7d3c0bf802640c78ac435531baff6432
(and also 4856d5a6b62836b9082a44da647aa01b0b89beb6 from #7192)

should be backported to Release 3.0 and 3.1 to create working installers.

Also note: by default GHActions is using a default NSIS 3.x, which does not support LargeStrings. This will cause our installers to fail to add stuff to $PATH during installation time on the target machine, if the PATH is reasonably filled.
Solution: wait for https://github.com/mkevenaar/chocolatey-packages/issues/221 or use the 'special build' of NSIS which supports 8k-strings.
--> let's use a custom NSIS (https://github.com/OpenMS/NSIS) in the meantime.

## Checklist

- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by timosachsenberg** [comment]
Also relevant for @JeeH-K

File: cmake/Windows/NSIS.template.in [hunk 1]
```diff
@@ -291,7 +284,7 @@ Section "TOPP tools" SEC_TOPP
     #open uninstall log file
     !insertmacro UNINSTALL.LOG_OPEN_INSTALL
 
-    # Since Qt 5.14 we probably do not need qt.conf if the plugins are in the standard location "plugins"
+    # Since Qt 5.14 we do not need qt.conf if the plugins are in the standard location "plugins"
     # Qt is built "-relocatable" since then.
     #File "${OPENMSBINDIR}\qt.conf"
```
Comment (jpfeuffer) [inline cmake/Windows/NSIS.template.in:289]: No surprise that older Qts don't find the plugins when this is commented out. But if Qt 5.15 is available everywhere we build installers I think I'm okay with requiring it.
But probably is not too hard to fix, either.

File: cmake/Windows/NSIS.template.in [hunk 2]
```diff
@@ -781,9 +652,9 @@ FunctionEnd
 !insertmacro MUI_DESCRIPTION_TEXT ${SEC_TOPP} "TOPP (The OpenMS PiPeline) - chainable tools for data analysis"
 !insertmacro MUI_DESCRIPTION_TEXT ${SEC_Doc} "Documentation/Tutorials for TOPP, TOPPView and the OpenMS library itself."
 !insertmacro MUI_DESCRIPTION_TEXT ${SEC_ThirdParty} "Install third party libraries (e.g., ProteoWizard)."
-!insertmacro MUI_DESCRIPTION_TEXT ${SEC_dotnet} "Install .NET and Microsoft Visual Studio Runtime for Proteowizard. (may require the installer to be run in admin mode). Deactivate if you are not admin and think this is installed already (or will be installed later) by an admin."
+#!insertmacro MUI_DESCRIPTION_TEXT ${SEC_dotnet} "Install .NET and Microsoft Visual Studio Runtime for Proteowizard. (may require the installer to be run in admin mode). Deactivate if you are not admin and think this is installed already (or will be installed later) by an admin."
```
Comment (jpfeuffer) [inline cmake/Windows/NSIS.template.in]: Remove?

File: cmake/Windows/NSIS.template.in [hunk 3]
```diff
@@ -781,9 +652,9 @@ FunctionEnd
...
+#!insertmacro MUI_DESCRIPTION_TEXT ${SEC_dotnet} "Install .NET and Microsoft Visual Studio Runtime for Proteowizard. (may require the installer to be run in admin mode). Deactivate if you are not admin and think this is installed already (or will be installed later) by an admin."
 !insertmacro MUI_DESCRIPTION_TEXT ${SEC_RegisterExt} "Register certain file types (e.g., '.mzML') with TOPPView.exe."
-!insertmacro MUI_DESCRIPTION_TEXT ${SEC_VSRuntime} "Install the required Microsoft Visual Studio runtime (may require the installer to be run in admin mode). Deactivate if you are not admin and think this is installed already (or will be installed later) by an admin."
+#!insertmacro MUI_DESCRIPTION_TEXT ${SEC_VSRuntime} "Install the required Microsoft Visual Studio runtime (may require the installer to be run in admin mode). Deactivate if you are not admin and think this is installed already (or will be installed later) by an admin."
```
Comment (jpfeuffer) [inline cmake/Windows/NSIS.template.in]: Remove?

**Review by jpfeuffer — APPROVED** [review]
Looks good. Small comments.
Did you test pwiz without the redists/.net?

Edit: Ah saw the PR description. You probably tested it..

File: cmake/Windows/NSIS.template.in (hunk 1 above)
Comment (cbielow) [reply cmake/Windows/NSIS.template.in:289]: jep. But if we can avoid the qt.conf, I'd avoid it.

**Comment by cbielow** [comment]
> Edit: Ah saw the PR description. You probably tested it..

I did :) (on a fresh Win10 VM)

**Pull request merged**